source/game/scripts/bind_threading.cpp

```diff
@@ -413,7 +414,8 @@ struct ScriptThread {
 		asIScriptFunction* localFunc = localMan->getFunction(func, "(double, ScriptThread&)", "double");
 
 		if(localFunc) {
-			if(runThread.compare_exchange_strong(2, 0) == 0) {
+			std::int32_t comparand = 0;
+			if(runThread.compare_exchange_strong(comparand, 2)) {
```

Comment:
the std::atomic version of compare_exchange_strong has the arguments switched, and returns bool instead of the value currently being stored. 

The first argument is also a reference, and has it's value replaced with the value stored in the atomic object at the time the operation is initiated. Previously this would have been the return value.